### ISSUE TITLE
TPLink EX 220 (AX1800)

### ISSUE THREAD

**Issue by metrotyranno** [sta-c0000/tpconf_bin_xml#26]
Hi,

This is unable to decrypt the config of the EX 220. Can someone help me get this config decrypted?

The script returns ERROR: Unrecognized file type!

I changed the script a bit so it returns the first 36 bytes:

b'\x83\xa1\x8d\x9ew\x84\xdb\x90M~\xc8\xecc\xa2\xa1\x16\xb9[\x9c\xd0o\x05\x90U@\xfc{\n\xa8)\x8f\xbf\x06\n\x17'

Can anyone help me get this decrypted?

**Comment by metrotyranno** [comment]
The device is basically an AX1800 with some ISP specific features. Would the GPL parts for the AX1800 offer any help?

**Comment by sta-c0000** [comment]
Ah, :) I thought that was a Wi-Fi 6 chipset qualifier (802.11ax / AX, 1800 Mbps).
I see now that TP-Link actually have an [AX1800](https://www.tp-link.com/us/support/download/archer-ax1800/) model.  Thanks.
I could be wrong, but because it is fairly recent, I still suspect it might be using a different method for config files.
Sorry, I can't see myself reverse engineering that firmware anytime soon.

**Comment by metrotyranno** [comment]
I managed to get my hands on the firmware bin. Would this be helpful in solving the issue?

**Comment by sta-c0000** [comment]
No, sorry, as I mentioned, I won't be personally reverse engineering those firmwares.
This project targets a class of TP-Link routers using similarly encoded configuration files; it wouldn't be useful with config files using a completely different storage format.
Cheers.

**Comment by metrotyranno** [comment]
See PR #58 :)

**Comment by sta-c0000** [comment]
What persistence!  Nice.
